### PR TITLE
Minor optimizations by using comprehensions

### PR DESCRIPTION
**Pull request by cclauss**
% `pipx install ruff`
% `ruff check --select=C4,PERF --statistics`
```
30	C408   	[*] Unnecessary `dict` call (rewrite as a literal)
10	C406   	[*] Unnecessary `tuple` literal (rewrite as a `dict` literal)
10	C417   	[*] Unnecessary `map` usage (rewrite using a generator expression)
 9	PERF401	[ ] Use a list comprehension to create a transformed list
 2	C419   	[*] Unnecessary list comprehension
 2	PERF203	[ ] `try`-`except` within a loop incurs performance overhead
 2	PERF402	[ ] Use `list` or `list.copy` to create a copy of a list
```
% `ruff check --select=C4,PERF --fix --unsafe-fixes`
```
Found 65 errors (52 fixed, 13 remaining).
```
% `ruff rule C408`
# unnecessary-collection-call (C408)

Derived from the **flake8-comprehensions** linter.

Fix is always available.

## What it does
Checks for unnecessary `dict`, `list` or `tuple` calls that can be
rewritten as empty literals.

## Why is this bad?
It's unnecessary to call, e.g., `dict()` as opposed to using an empty
literal (`{}`). The former is slower because the name `dict` must be
looked up in the global scope in case it has been rebound.

## Examples
```python
dict()
dict(a=1, b=2)
list()
tuple()
```

Use instead:
```python
{}
{"a": 1, "b": 2}
[]
()
```

## Fix safety
This rule's fix is marked as unsafe, as it may occasionally drop comments
when rewriting the call. In most cases, though, comments will be preserved.

## Options
- `lint.flake8-comprehensions.allow-dict-calls-with-keyword-arguments`

### CODE REVIEW

File: buildconfig/config.py [hunk 1]
```diff
@@ -80,14 +80,12 @@ def prepdep(dep, basepath):
...
-            for dir in dep.inc_dir:
-                incs.append(IPREFIX+dir[startind:])
+            incs.extend(IPREFIX+dir[startind:] for dir in dep.inc_dir)
     if dep.lib_dir:
         if isinstance(dep.lib_dir, str):
             lids.append(LPREFIX+dep.lib_dir[startind:])
         else:
-            for dir in dep.lib_dir:
-                lids.append(LPREFIX+dir[startind:])
+            lids.extend(LPREFIX+dir[startind:]for dir in dep.lib_dir)
```
Comment (ankith26) [inline buildconfig/config.py:88]:
```suggestion
            lids.extend(LPREFIX+dir[startind:] for dir in dep.lib_dir)
```
formatting issue, must have space before `for`

**Review by ankith26 — CHANGES_REQUESTED** [review]
The lint job seems to have failed with other errors on this PR, so a commit fixing all those will also be needed

**Comment by cclauss** [comment]
Are you talking about #2773 or something else?
* #2773

**Comment by ankith26** [comment]
I'm talking about this fail: https://github.com/pygame-community/pygame-ce/actions/runs/8425378679/job/23088593026

**Review by ankith26 — APPROVED** [review]
LGTM, thanks for the PR :partying_face:

**Comment by Starbuck5** [comment]
This PR now has merge conflicts.

**Comment by cclauss** [comment]
Git conflicts fixed.

**Review by zoldalma999 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**